tests/expose_test.go

```diff
@@ -0,0 +1,104 @@
+package tests_test
+
+import (
+	"flag"
+	"strings"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"kubevirt.io/kubevirt-ansible/tests"
+	"kubevirt.io/kubevirt/pkg/kubecli"
+	ktests "kubevirt.io/kubevirt/tests"
+)
+
+const (
+	vmName       = "vm-cirros"
+	testPod      = "testpod"
+	vmCirrosPath = "tests/manifests/vm-cirros.yml"
+)
+
+var _ = Describe("VM/VMI system test", func() {
+
+	flag.Parse()
+	virtClient, err := kubecli.GetKubevirtClient()
+	ktests.PanicOnError(err)
+
+	Context("Expose service via ssh and console VNC", func() {
+
+		It("the VM should be created successfully", func() {
+			args := []string{"create", "-f", vmCirrosPath, "-n", tests.NamespaceTestDefault}
+			_, err := ktests.RunCommand("oc", args...)
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		It("the VM should be started by virtctl", func() {
+			args := []string{"start", vmName, "-n", tests.NamespaceTestDefault}
+			_, err := ktests.RunCommand("virtctl", args...)
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		Specify("the VMI should be running", func() {
+			By("Wait until vmi ready")
+			Eventually(func() bool {
+				args := []string{"get", "vmi", vmName, "-o=jsonpath='{.status.phase}'", "-n", tests.NamespaceTestDefault}
+				out, err := ktests.RunCommand("oc", args...)
+				Expect(err).ToNot(HaveOccurred())
+				return strings.Contains(out, "Running")
+			}, time.Duration(2)*time.Minute).Should(BeTrue(), "Timed out waiting for vmi to appear")
+		})
+
+		It("the VM should be exposed by virtctl via ClusterIP", func() {
+			args := []string{"expose", "vmi", vmName, "-n", tests.NamespaceTestDefault, "--name=myvm-ssh", "--port=22", "--type=ClusterIP"}
+			_, err := ktests.RunCommand("virtctl", args...)
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		It("the VMI should be able to connect via ssh", func() {
+			By("Create a pod as ssh client to connect to VMI")
+			args := []string{"run", testPod, "-n", tests.NamespaceTestDefault, "--image=shiywang/test-ssh", "--", "sleep", "3600"}
```

Comment:
what's shiywang/test-ssh?

Reply:
ping

Reply:
it's because the testing vm was exposed by ClusterIP, which either you execute ssh on the node, or start a pod to access it, so shiywang/test-ssh is the pod I created to acesss VM via ssh, I believe we can move this to kubevirt repo, but I don't have any permissions